### ISSUE TITLE
Bug: Unable to remove invitations again `confluent_invitation`

### ISSUE THREAD

**Issue by Gaardsholt** [confluentinc/terraform-provider-confluent#263]
I have the following terraform:
```hcl
variable "members" {
  type = list(string)
  default = [
    "firstname.lastname@example.com",
    "john.doe@example.com"
  ]
}
resource "confluent_invitation" "main" {
  for_each  = toset(var.members)
  email     = each.value
  auth_type = "AUTH_TYPE_SSO"
}
```

and that works fine and the two users are being invited.

If I then try and remove one of the users I get this error:
```
confluent_invitation.main["firstname.lastname@example.com"]: Destroying... [id=i-16926]
╷
│ Error: error deleting invitation "i-16926": 400 Bad Request: Bad Request. The invitation was already accepted
│ 
│ 
╵
```

**Comment by linouk23** [comment]
👋 @Gaardsholt, thanks for creating the issue!

It seems to be a known limitation that kinda makes sense since UI would still display this invitation as existing.

Would you recommend having a "fake" delete for this resource still so you could make your TF state empty or something?

**Comment by Gaardsholt** [comment]
Basically I just want a way to manage the users in my organization.
So maybe just a `confluent_user` that can handle both the invitation and the user object after the invitation has been accepted.
That way I can invite our users and remove them later on if they leave the organization or doesn't need the access anymore

**Comment by minionOfZuul** [comment]
This limitation, and the lack of a confluent_user resource make it really obnoxious to manage user accounts for a large and constantly changing pool of users.

**Comment by linouk23** [comment]
Thanks for the comments!

We discussed this UX issue internally and agreed that for now TF Provider will have to mirror our API / UI behavior.

That said, for the next iteration of Invitations API hopefully we'll be able to apply the suggestions above, thanks!

**Comment by linouk23** [comment]
@Gaardsholt @minionOfZuul will it be helpful to add a TF only boolean attribute like `allow_deletion` that'd delete an already accepted invitation instances from TF state?

**Comment by minionOfZuul** [comment]
Until the provider has a user resource that we can delete, we'll probably just hook our user provisioning/management up via a lambda function or something. Thanks, though.

**Comment by harideveloper** [comment]
+1 to this issue. Reading the previous comments I understand the API/UX behaviour is replicated, But It'd good to have users & user permissions managed through IAC. I'd appreciate if we could revisit this issue, alternatively views on possible  workaround / temporary fix to go through issues could also help. Thanks

**Comment by Gaardsholt** [comment]
Any updates on this?
Do you need help doing this?
I would be happy to take a crack at it.

**Comment by linouk23** [comment]
@Gaardsholt there are no updates yet. We're trying to see how many users would be interested in this feature to prioritize it accordingly. If you could help with implementation, that'd be great!

**Comment by dcbagger** [comment]
> @Gaardsholt there are no updates yet. We're trying to see how many users would be interested in this feature to prioritize it accordingly. If you could help with implementation, that'd be great!

Where can we declare that we would like this feature to prioritized?

**Comment by linouk23** [comment]
> @Gaardsholt there are no updates yet. We're trying to see how many users would be interested in this feature to prioritize it accordingly. If you could help with implementation, that'd be great! Where can we declare that we would like this feature to prioritized?

@dcbagger That's a great question! Upvoting the [first](https://github.com/confluentinc/terraform-provider-confluent/issues/263#issue-1720506976) message in this thread should be a great start, thanks!

**Comment by alexvanderberkel** [comment]
What would be the threshold you need to get this implemented?

**Comment by linouk23** [comment]
@alexvanderberkel that's a great question! Fifteen upvotes do seem like a lot. I will reach out to our product team again to discuss prioritization. For the context, there're 2 paths forward:

* low LOE: https://github.com/confluentinc/terraform-provider-confluent/issues/263#issuecomment-1601558298 -- but based on number of upvotes, it didn't look good enough
* high LOE: release a new major API version, update UX in all clients

**Comment by tmalikconfluent** [comment]
fixed in v1.81.0 terraform release
